src/components/ActivityCollector/getLinkName.js

```diff
@@ -0,0 +1,130 @@
+    supportedNodes.push(node);
+    if (node.childNodes) {
+      const childNodes = Array.prototype.slice.call(node.childNodes);
+      childNodes.forEach(childNode => {
+        const nodes = extractSupportedNodes(childNode);
+        supportedNodes = supportedNodes.concat(nodes.supportedNodes);
+        includesUnsupportedNodes =
+          includesUnsupportedNodes || nodes.includesUnsupportedNodes;
+      });
+    }
+  } else {
+    includesUnsupportedNodes = true;
+  }
+  return {
+    supportedNodes,
+    includesUnsupportedNodes
+  };
+};
+
+/**
+ * Returns the value of a node attribute.
+ * @param {*} node The node holding the attribute.
+ * @param {string} attributeName The name of the attribute.
+ * @param {string} nodeName Optional node name constraint.
+ * @returns {string} Attribute value or undefined.
+ */
+const getNodeAttributeValue = (node, attributeName, nodeName) => {
+  let attributeValue;
+  if (!nodeName || nodeName === node.nodeName.toUpperCase()) {
+    attributeValue = node.getAttribute(attributeName);
+  }
+  return attributeValue;
+};
+
+/**
+ * Extracts a link-name from a given node.
+ *
+ * The returned link-name is set to one of the following (in order of priority):
+ *
+ * 1. Clicked node innerText
+ * 2. Clicked node textContent
+ * 3. Clicked node and its child nodes nodeValue appended together.
+ * 4. Clicked node alt attribute or node descendant alt attribute.
+ *    Whichever is found first.
+ * 5. Clicked node text attribute or node descendant text attribute.
+ *    Whichever is found first.
+ * 6. Clicked node INPUT descendant value attribute.
+ *    Whichever is found first.
+ * 7. Clicked node IMG descendant src attribute.
+ *    Whichever is found first.
+ *
+ * @param {*} node The node to find link text for.
+ * @returns {string} link-name or an empty string if not link-name is found.
+ */
+export default node => {
+  let nodeText = truncateWhiteSpace(node.innerText || node.textContent);
+  const nodes = extractSupportedNodes(node);
+  if (!nodeText || nodes.includesUnsupportedNodes) {
+    let alt;
+    let title;
+    let inputValue;
+    let imgSrc;
+    const texts = [];
+    nodes.supportedNodes.forEach(supportedNode => {
+      if (supportedNode.getAttribute) {
+        alt = alt || truncateWhiteSpace(supportedNode.getAttribute("alt"));
+        title =
+          title || truncateWhiteSpace(supportedNode.getAttribute("title"));
+        inputValue =
+          inputValue ||
+          truncateWhiteSpace(
+            getNodeAttributeValue(supportedNode, "value", "INPUT")
+          );
+        imgSrc =
+          imgSrc ||
+          truncateWhiteSpace(
+            getNodeAttributeValue(supportedNode, "src", "IMG")
+          );
+      }
+      if (supportedNode.nodeValue) {
+        texts.push(supportedNode.nodeValue);
+      }
+    });
+    nodeText = truncateWhiteSpace(texts.join(""));
+    if (!nodeText) {
+      nodeText = truncateWhiteSpace(alt || title || inputValue || imgSrc || "");
```

Comment:
Why not call `truncateWhiteSpace` in this one spot only?

Reply:
we'd like to avoid collecting values with spaces only